worlds/ahit/__init__.py

```diff
@@ -0,0 +1,356 @@
+                    full_clear.show_in_spoiler = False
+
+            return
+
+        if self.options.ActRandomizer.value > 0:
+            randomize_act_entrances(self)
+
+        set_rules(self)
+
+        if self.is_dw():
+            set_dw_rules(self)
+
+    def create_item(self, name: str) -> Item:
+        return create_item(self, name)
+
+    def fill_slot_data(self) -> dict:
+        slot_data: dict = {"Chapter1Cost": chapter_timepiece_costs[self.player][ChapterIndex.MAFIA],
+                           "Chapter2Cost": chapter_timepiece_costs[self.player][ChapterIndex.BIRDS],
+                           "Chapter3Cost": chapter_timepiece_costs[self.player][ChapterIndex.SUBCON],
+                           "Chapter4Cost": chapter_timepiece_costs[self.player][ChapterIndex.ALPINE],
+                           "Chapter5Cost": chapter_timepiece_costs[self.player][ChapterIndex.FINALE],
+                           "Chapter6Cost": chapter_timepiece_costs[self.player][ChapterIndex.CRUISE],
+                           "Chapter7Cost": chapter_timepiece_costs[self.player][ChapterIndex.METRO],
+                           "BadgeSellerItemCount": badge_seller_count[self.player],
+                           "SeedNumber": str(self.multiworld.seed),  # For shop prices
+                           "SeedName": self.multiworld.seed_name,
+                           "TotalLocations": get_total_locations(self)}
+
+        if self.options.HatItems.value == 0:
+            slot_data.setdefault("SprintYarnCost", hat_yarn_costs[self.player][HatType.SPRINT])
+            slot_data.setdefault("BrewingYarnCost", hat_yarn_costs[self.player][HatType.BREWING])
+            slot_data.setdefault("IceYarnCost", hat_yarn_costs[self.player][HatType.ICE])
+            slot_data.setdefault("DwellerYarnCost", hat_yarn_costs[self.player][HatType.DWELLER])
+            slot_data.setdefault("TimeStopYarnCost", hat_yarn_costs[self.player][HatType.TIME_STOP])
+            slot_data.setdefault("Hat1", int(hat_craft_order[self.player][0]))
+            slot_data.setdefault("Hat2", int(hat_craft_order[self.player][1]))
+            slot_data.setdefault("Hat3", int(hat_craft_order[self.player][2]))
+            slot_data.setdefault("Hat4", int(hat_craft_order[self.player][3]))
+            slot_data.setdefault("Hat5", int(hat_craft_order[self.player][4]))
+
+        if self.options.ActRandomizer.value > 0:
+            for name in self.act_connections.keys():
+                slot_data[name] = self.act_connections[name]
+
+        if self.is_dlc2() and not self.is_dw_only():
+            for name in nyakuza_thug_items[self.player].keys():
+                slot_data[name] = nyakuza_thug_items[self.player][name]
+
+        if self.is_dw():
+            i: int = 0
+            for name in excluded_dws[self.player]:
+                if self.options.EndGoal.value == 3 and name == "Seal the Deal":
+                    continue
+
+                slot_data[f"excluded_dw{i}"] = dw_classes[name]
```

Comment:
The reason why everything in the slot data is a dict is because it makes it much easier to handle on the mod's end. JSON in UnrealScript has no concept of square brackets, and therefore lists, so making everything into a dict is much easier.

Reply:
```
for (i = 0; i <= 99; i++)
{
	n = json.GetStringValue("excluded_dw"$i);
	if (n == "")
		break;
	
	dw = class<Hat_SnatcherContract_DeathWish>(class'Hat_ClassHelper'.static.ClassFromName(n));
	SlotData.ExcludedContracts.AddItem(dw);
}